LibreHardwareMonitorLib/Hardware/Motherboard/Lpc/EC/EmbeddedController.cs

```diff
@@ -6,20 +6,151 @@
 using System;
 using System.Collections.Generic;
 using System.Globalization;
+using System.Linq;
 using System.Text;
 
 namespace LibreHardwareMonitor.Hardware.Motherboard.Lpc.EC
 {
     public abstract class EmbeddedController : Hardware
     {
+        enum ECSensor
+        {
+            /// <summary>chipset temperature [℃]</summary>
+            TempChipset,
+            /// <summary>CPU temperature [℃]</summary>
+            TempCPU,
+            /// motherboard temperature [℃]</summary>
+            TempMB,
+            /// <summary>"T_Sensor" temperature sensor reading [℃]</summary>
+            TempTSensor,
+            /// <summary>VRM temperature [℃]</summary>
+            TempVrm,
+            /// <summary>CPU_Opt fan [RPM]</summary>
+            FanCPUOpt,
+            /// <summary>VRM heat sink fan [RPM]</summary>
+            FanVrmHS,
+            /// <summary>Chipset fan [RPM]</summary>
+            FanChipset,
+            /// <summary>Water flow sensor reading [RPM]</summary>
+            FanWaterFlow,
+            /// <summary>CPU current [A]</summary>
+            CurrCPU,
+            /// <summary>"Water_In" temperature sensor reading [℃]</summary>
+            TempWaterIn,
+            /// <summary>"Water_Out" temperature sensor reading [℃]</summary>
+            TempWaterOut,
+            Max
+        };
+
+        private static readonly Dictionary<ECSensor, EmbeddedControllerSource> _knownSensors = new()
+        {
+            { ECSensor.TempChipset, new EmbeddedControllerSource("Chipset", SensorType.Temperature, 0x003a) },
+            { ECSensor.TempCPU, new EmbeddedControllerSource("CPU", SensorType.Temperature, 0x003b) },
+            { ECSensor.TempMB, new EmbeddedControllerSource("Motherboard", SensorType.Temperature, 0x003c) },
+            { ECSensor.TempTSensor, new EmbeddedControllerSource("T_Sensor", SensorType.Temperature, 0x003d, blank: -40) },
+            { ECSensor.TempVrm, new EmbeddedControllerSource("VRM", SensorType.Temperature, 0x003e) },
+            { ECSensor.FanCPUOpt, new EmbeddedControllerSource("CPU_Opt", SensorType.Fan, 0x00b0, 2) },
+            { ECSensor.FanVrmHS, new EmbeddedControllerSource("VRM HS", SensorType.Fan, 0x00b2, 2) },
+            { ECSensor.FanChipset, new EmbeddedControllerSource("Chipset", SensorType.Fan, 0x00b4, 2) },
+            // TODO: "why 42?" is a silly question, I know, but still, why? On the serious side, it might be 41.6(6)
+            { ECSensor.FanWaterFlow, new EmbeddedControllerSource("Water_Flow", SensorType.Flow, 0x00bc, 2, factor: 1.0f / 42f * 60f) },
+            { ECSensor.CurrCPU, new EmbeddedControllerSource("CPU", SensorType.Current, 0x00f4) },
+            { ECSensor.TempWaterIn, new EmbeddedControllerSource("Water_In", SensorType.Temperature, 0x0100, blank: -40) },
+            { ECSensor.TempWaterOut, new EmbeddedControllerSource("Water_Out", SensorType.Temperature, 0x0101, blank: -40) },
```

Comment:
Can you keep the original naming here, i.e. T Sensor, Water Flow, Water In, and Water Out.

Reply:
Oops, changed labels by mistake.

Reply:
Also CPU_Opt, might as well make it CPU Optional :)